clang/lib/CodeGen/CGExprComplex.cpp

```diff
@@ -347,6 +355,8 @@ class ComplexExprEmitter
 #define HANDLEBINOP(OP)                                                        \
   ComplexPairTy VisitBin##OP(const BinaryOperator *E) {                        \
     QualType promotionTy = getPromotionType(                                   \
+        (E->hasStoredFPFeatures() ? E->getStoredFPFeatures()                   \
```

Comment:
This pattern comes up often enough, I wonder if we should add `getStoredFPFeaturesOrDefault()` that does this dance for us?